proto/storage/node.proto

```diff
@@ -86,11 +89,17 @@ message NodeScan {
 }
 
 message NodeInventory {
-    string                         node_id    = 1 [(gogoproto.moretags) = 'search:"Node ID,store" sql:"pk,type(uuid)"'];
-    string                         node_name  = 2 [(gogoproto.moretags) = 'search:"Node,store"'];
-    google.protobuf.Timestamp      scan_time  = 3 [(gogoproto.moretags) = 'search:"Node Scan Time,store"'];
-    scannerV1.Components           components = 4;
-    repeated scannerV1.Note        notes      = 5;
```

Comment:
This is correct, the `NodeInventory` is new and we allow ourselves a bit more changes. But I followed your suggestion and kept the reserved numbers.

Reply:
I dig into the protoloc (the binary that checks proto backwards compatibility) and there are two options that we can do:

1. Overwrite the proto.lock file forcing any incompatibilities to be ignored - should be no harm as there are 0 clusters running old version of Compliance but new version of Sensor (both are using the same image). This would  allow us to get rid of the `reserved` fields and write any definition of NodeInventory we want.
2. Keep the strict backwards compatibility and reserve the numbers 1 and 4 but also names `node_id` and `components` ([why names also matter for protolock's author](https://github.com/nilslice/protolock/issues/52#issuecomment-431084318)). This would require us to rename the field `components` and start reserving the names additionally to the numbers.

In the given situation, I tend to choose the first option. If there are any opinions against this, I am fine to revisit this choice. (FYI: @rhybrillou )

Reply:
Considering `NodeInventory` is being added to `Node` here, I don't expect any `storage.NodeInventory` with the old format present in the store or on the communication channels.
Should there be the slightest risk of our processes exploding in flight while decoding objects encoded with the old format, I'd go for the more conservative approach.
In the current state of things, I think the easier first option is OK. Please make sure the `proto.lock` file is also updated accordingly.